src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,816 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 19 June 2018
+
+@author: Anders Muskens
+
+Copyright © 2018 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division
+from concurrent.futures import CancelledError, TimeoutError
+import fcntl
+import glob
+import serial
+import logging
+import os
+import threading
+import time
+import re
+from Pyro4.core import isasync
+from odemis import model
+from odemis.util import driver
+from odemis.model import HwError, CancellableFuture, ParallelThreadPoolExecutor
+
+# Unit definitions 
+UNIT_DEF = {
+    'mm': 2,
+    'um': 3,
+    'in': 4,
+    }
+
+C = 2.99792458e11  # speed of light in mm/s
+
+
+def estimateTimeForMove(d, v, ac, dc):
+    """
+    Calcualtes the time for a  move of d given speed v, acceleration (ac) and deceleration (dc)
+    d (float) in mm
+    v (float) in mm/s
+    ac, dc (float) in mm/s^2
+    """
+
+    # time and distance of the acceleration to speed
+    t_ac = v / ac
+    d_ac = 0.5 * ac * t_ac ** 2
+
+    # time and distance of the decelerated portion
+    t_dc = v / dc
+    d_dc = 0.5 * dc * t_dc ** 2
+
+    # distance  and time at uniform motion
+    d_uni = d - d_ac - d_dc
+    t_uni = d_uni / v
+
+    return t_ac + t_dc + t_uni
+
+
+class ESPError(Exception):
+    """
+    Exception used to indicate a problem reported by the device.
+    """
+    pass
+
+
+class ESP(model.Actuator):
```

Comment:
Here you can document the driver in general (ie, the first two lines of the __init__ docstring). Also mention here that MD_POS_COR can be used to set an offset (and that it should be a dict).